pubsub/natspubsub/nats.go

```diff
@@ -93,12 +117,15 @@ func init() {
 // defaultDialer dials a default NATS server based on the environment
 // variable "NATS_SERVER_URL".
 type defaultDialer struct {
-	init   sync.Once
-	opener *URLOpener
-	err    error
+	init sync.Once
+	conn *nats.Conn
```

Comment:
Do we need this? I see you setting it on line 140, but I don't think it's ever used...?

Reply:
That's true, the `conn` is used only during the `sync.Once.Do` and it is passed to both url openeres, so there is no need to keep it in the `defaultDialer`.